### ISSUE TITLE
[7.0.0 & 8.1] Searching not working (properly) in Chinese language

### ISSUE THREAD

**Issue by KKyang** [owncloud/core#9916]
### System Configuration

**OS:** Ubuntu 14.04 LTS with desktop environment
**Server:** Apache 2.4.7
**PHP version:** 5.5.9
**Database:** MySQL
**OwnCloud:** 7.0.0 update from 6.0.4
### Step to Reproduce
1. Try to search "口試"
2. Nothing happened
   ![78](https://cloud.githubusercontent.com/assets/2676255/3709971/7a5164e8-146a-11e4-9475-e0d0ab5f5e89.png)
3. But if I press left arrow, the files related will be highlighted
   ![79](https://cloud.githubusercontent.com/assets/2676255/3709973/954f2780-146a-11e4-9f8f-7c518ca2e889.png)
4. If I search "open", the relared files will be highlighted immediately and a search result will pop up.
   ![80](https://cloud.githubusercontent.com/assets/2676255/3709979/d132fa88-146a-11e4-8996-f65977fc432f.png)
### Actual Behavior

Searching not working properly, see above.
### Expect Behavior

The files are highlighted
### Owncloud log

No log shown.

**Comment by Xenopathic** [comment]
@butonic What is the progress on this issue? Is it still valid, or is this behaviour expected?

**Comment by PVince81** [comment]
@dttpy can you check this on 7.0.5 / 8.0.2 ?

**Comment by butonic** [comment]
In OC8 we filter files instead of highlighting them. Tested on sqlite with `口試同意書.txt` and searching for 口試. Adding the same file to subfolders also returns them as results in other places. 

@Xenopathic @KKyang if you can confirm we can close this.

**Comment by dttpy** [comment]
I've tested it with 7.0.4 / 8.0.2 and master, and OC7 and OC8 have different behaviours.
With OC7, testing like the report, searching for chineses file name doesn't work, because it need at least 3 characters in the searching field. If the file name isn't long enough like "口试.txt", in order to find the file, the rest part of the file name ".txt" should also be given.  
With OC8, if the file with chinese name is directly under the searched folder / root, the search works well. It just need one character and can find the proper file. But if this file is in some other foldere, it needs at least 3 characters in the search name, so that it can start to search in other folders, which is the same by searching the files with alphabet / number / etc. in the name. 
To summarize: searching works in chinese language, but it is not optimal, because of the 3 characters limit. Chinese language is more compact than the western language. One character means already much, and 2 characters can already be a word. So 3 is not so convenient for the chinese speaking user, because not many of them will try so much.
@PVince81

**Comment by KKyang** [comment]
Sorry for replying this thread so late.

Recently I upgrade OwnCloud to 8.1 (from 8.0.5)

But there's still a problem, if I search 助教資料
![277](https://cloud.githubusercontent.com/assets/2676255/8694714/611a47e0-2b13-11e5-963b-5305244c19b5.png)

I get one searching result. Then  I search 助教資 (first three words instead of four)
![278](https://cloud.githubusercontent.com/assets/2676255/8694723/6e717030-2b13-11e5-8291-4337dc2f52e9.png)

I still get one result, but when i search 助教
![280](https://cloud.githubusercontent.com/assets/2676255/8694742/9ae82a46-2b13-11e5-9a80-603bf12b690a.png)

and

![279](https://cloud.githubusercontent.com/assets/2676255/8694746/a1aff69c-2b13-11e5-9083-0c44fc2713ac.png)

No result is returned. Is this normal or is this still a bug? Thanks!

**Comment by ringceving** [comment]
Issue still exist in version 9.0.2.

**Comment by butonic** [comment]
@ringceving can I ask you to try and report on the workaround as in https://github.com/owncloud/core/issues/25021#issuecomment-224526597

**Comment by hodyroff** [comment]
How does 10.0 Beta look like for you?

**Comment by hodyroff** [comment]
No further feedback. Closing.
